sdk/iot/pnp/src/az_iot_pnp_client.c

```diff
@@ -53,32 +54,42 @@ AZ_NODISCARD az_result az_iot_pnp_client_init(
 
 AZ_NODISCARD az_result az_iot_pnp_client_get_user_name(
     az_iot_pnp_client const* client,
-    az_span mqtt_user_name,
-    az_span* out_mqtt_user_name)
+    char* mqtt_user_name,
+    size_t mqtt_user_name_size,
+    size_t* out_mqtt_user_name_length)
 {
   AZ_PRECONDITION_NOT_NULL(client);
-  AZ_PRECONDITION_VALID_SPAN(mqtt_user_name, 0, false);
-  AZ_PRECONDITION_NOT_NULL(out_mqtt_user_name);
+  AZ_PRECONDITION_NOT_NULL(mqtt_user_name);
+  AZ_PRECONDITION(mqtt_user_name_size > 0);
 
-  az_span remainder = mqtt_user_name;
-  az_span user_name;
-  AZ_RETURN_IF_FAILED(
-      az_iot_hub_client_user_name_get(&client->_internal.iot_hub_client, remainder, &user_name));
+  size_t written;
 
-  int32_t written = az_span_size(user_name);
-  remainder = az_span_slice_to_end(remainder, written);
+  // First get hub user name since it is unknown how long it will be
```

Comment:
"since it is unknown how long it will be" - I think we should remove this.  We need to make this call in any case because we're writing this user name into the `mqtt_user_name` to build up protocol.  That segment of wording looks like we're just doing a length check but throwing away the results.

Reply:
Similar comment in the PnP telemetry file with same verbiage.  I'm promoting this beyond a nit:), because it kind of threw my code read at 1st.

Reply:
okay understood. The original intent of the comment was to say that we must get this part before going to calculate the required size for pnp.